### PR TITLE
Add examples for lexer / parser files to main README

### PR DESCRIPTION
**Pull request by pablosichert**
Hey there, I think examples for the actual lexer and parser files on the landing page could be valuable (although they add a small burden of needing to be kept up-to-date).

I copied those from [here](https://github.com/softdevteam/grmtools/blob/edeb6fafa613d64c05d7f282a1616220df6baf44/lrpar/README.md) and added Rust syntax hightlighting (because I think it works exceptionally well with those).

#133

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,8 +29,59 @@ fn main() -> Result<(), Box<std::error::Error>> {
 }
 ```
 
-This will generate and compile a parser and lexer using the definitions found
-in `src/lexer.l` and `src/grammar.y`. We can then use the generated lexer and
+This will generate and compile a parser and lexer, where the definitions for the lexer can be found in
```
Comment (ltratt) [inline README.md]: Can we word wrap this line and join the `src/calc.l` filename to it?

File: README.md [hunk 2]
```diff
@@ -29,8 +29,59 @@ fn main() -> Result<(), Box<std::error::Error>> {
...
+        Ok(val) => Ok(val),
+        Err(_) => {
+            eprintln!("{} cannot be represented as a u64", s);
+            Err(())
+        }
+    }
+}
+```
+
+We can then use the generated lexer and
```
Comment (ltratt) [inline README.md]: Similarly, let's join this line with the one below it (and word wrap).

**Comment by ltratt** [comment]
@pablosichert I hadn't noticed there were new commits in this PR -- sorry!

File: README.md [hunk 3]
```diff
@@ -29,9 +29,57 @@ fn main() -> Result<(), Box<std::error::Error>> {
 }
 ```
 
-This will generate and compile a parser and lexer using the definitions found
-in `src/lexer.l` and `src/grammar.y`. We can then use the generated lexer and
-parser within our `src/main.rs` file as follows:
+This will generate and compile a parser and lexer, where the definitions for the lexer can be found in `src/calc.l`:
```
Comment (ltratt) [inline README.md]: Can we wrap this line to 80 chars?

File: README.md [hunk 4]
```diff
@@ -29,9 +29,57 @@ fn main() -> Result<(), Box<std::error::Error>> {
...
+        Ok(val) => Ok(val),
+        Err(_) => {
+            eprintln!("{} cannot be represented as a u64", s);
+            Err(())
+        }
+    }
+}
+```
+
+We can then use the generated lexer and parser within our `src/main.rs` file as follows:
```
Comment (ltratt) [inline README.md]: Can we wrap this line to 80 chars?

**Comment by ltratt** [comment]
@pablosichert Sorry for the slow reply. If you could send a message when you've pushed new commits, it would be really cool -- unfortunately I don't get notified otherwise :(

**Comment by pablosichert** [comment]
No worries!

I just applied your suggestions and squashed the commits

**Pull request merged**